Language/Expressions/Identifier_Reference/built_in_identifier_A26_t02.dart

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion It is a syntax error if a built-in identifier is used as the
+/// declared name of a prefix, class, mixin, enum, type parameter, type alias,
+/// or extension.
+///
+/// @description Checks that it is a compile-time error if a built-in identifier
+/// `augment` is used as the declared name of a type variable.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
+
+class C<augment> {
+//      ^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
+
+mixin M<augment> {
+//      ^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
+
+enum E<augment> {
+//     ^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+  e1;
+}
+
+void foo<augment>() {}
+//       ^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+
+extension Ext<augment> on List {}
+//            ^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+
```

Comment:
We could also have a type variable declared by a type alias or by a function type:
```dart
typedef int F1<augment>();
typedef F2<augment extends Comparable<augment>> = int Function();

const void Function<augment>()? c = null;
```